objects/src/notes/mod.rs

```diff
@@ -29,11 +29,13 @@ pub use inputs::NoteInputs;
 mod metadata;
 pub use metadata::NoteMetadata;
 
+pub mod note_execution_hint;
```

Comment:
Is there a reason to make the whole submodule public rather than re-export relevant components from it as we do with other components here?

Reply:
Changed it to only export the enum! Was keeping everything public until out of draft